lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GSMobileIdentity.cpp

```diff
@@ -230,9 +259,26 @@ int M5GSMobileIdentityMsg::DecodeImsiMobileIdentityMsg(
                << int(imsi->protect_schm_id);
   MLOG(MDEBUG) << "  Home Network Public Key Identifier = " << std::hex
                << int(imsi->home_nw_id);
-  MLOG(MDEBUG) << "  Scheme Output = ";
-  BUFFER_PRINT_LOG(imsi->scheme_output, tmp)
 
+  if ((imsi->type_of_identity == M5GSMobileIdentityMsg_SUCI_IMSI) &&
+      (imsi->protect_schm_id != 0)) {
+    if (imsi->protect_schm_id == PROFILE_A) {
+      MLOG(MDEBUG) << "  ECC empheral public key = ";
+      BUFFER_PRINT_LOG(imsi->empheral_public_key, EPHEMERAL_PUBLIC_KEY_LENGTH)
+    }
+    if (imsi->protect_schm_id == PROFILE_B) {
+      MLOG(MDEBUG) << "  ECC empheral public key = ";
+      BUFFER_PRINT_LOG(
+          imsi->empheral_public_key,
+          EPHEMERAL_PUBLIC_KEY_LENGTH + PROFILE_B_LEN)
+    }
+
+    MLOG(MDEBUG) << "  Ciphertext = ";
+    BUFFER_PRINT_LOG(imsi->ciphertext, CIPHERTEXT_LENGTH)
+
+    MLOG(MDEBUG) << "  MAC tag = ";
+    BUFFER_PRINT_LOG(imsi->mac_tag, MAC_TAG_LENGTH)
+  }
```

Comment:
don't dump chunks of code for debugging correct encoding/decoding purposes. Use unit tests to verify instead for each of these cases.

Reply:
can you go through this file and revisit (and remove) all such prints in encoding/decoding functions that are there with the sole purpose of code debugging during development phase.

Reply:
Ok, removed these debug logs as the UT is already written to verify the encode/decode data.